src/components/graphs/AnnualCycleGraph/AnnualCycleGraph.js

```diff
@@ -129,28 +205,38 @@ export default class AnnualCycleGraph extends React.Component {
     exportDataToWorksheet(
       'timeseries',
       _.pick(this.props, 'model_id', 'variable_id', 'experiment', 'meta'),
-      this.state.graphSpec,
+      this.graphSpec(),
```

Comment:
So the graph is rebuilt from API results if someone wants to download a file? Have you tested about how long that takes? I assume it's pretty fast - the slow part is fetching the data in the first place.

Reply:
Good question. Happily, by design, no backend API calls are made:
`graphSpec` (when it has data) calls 
- `this.getMetadatas`, which in turn calls `props.getMetadata`, which is a function that returns a filtered subset of the already-fetched metadata. 
  - The components which specialize ACG (by passing in `props.getMetadata`) do not (or at least should not) hit the backend to obtain this metadata; they already have it (passed in as a prop). 
  - Metdata is only obtained from the backend when the model, emissions, or variable selection changes. This is the responsibility of the app controllers, way up the component tree. Metadata passes as props from there down the tree. 
- `props.dataToGraphSpec`, which mixes up a batch of special objects to please C3. This is based on 
  - results of `this.getMetadatas` above
  - `state.data` (which is gotten from the backend when and only when the metadata, area, or current dataSpec changes). No (extra) backend calls here either.

Reply:
Hmm, I wonder if this explanation belongs in comments somewhere.